gdax/public_client.py

```diff
@@ -8,69 +8,165 @@
 
 
 class PublicClient(object):
-    def __init__(self, api_url="https://api.gdax.com", product_id="BTC-USD"):
-        self.url = api_url.rstrip("/")
+    """GDAX public client API.
+
+    All requests default to the `product_id` specified at object creation if not otherwise specified.
+
+    Attributes:
+        url (str): API endpoint URL
+        product_id (str): Default product used for requests
+
+    """
+
+    def __init__(self, api_url='https://api.gdax.com', product_id='BTC-USD'):
```

Comment:
This is a good point. I originally didn't change this out of inertia (and the node.js implementation has the same behavior) but honestly it should change. I see two options:
* require specifying `product_id` for every function.
* Keep `product_id` as an optional parameter: `def __init__(self, api_url='https://api.gdax.com', product_id=None)`. If a `product_id` is specified on object initialization, make that the default value for the object. If no default `product_id` is specified, require that one be specified.

Any thoughts? It's a small trade-off between convenience and explicitness.